### PR TITLE
Add commlib-py

### PR DESCRIPTION
**Pull request by MementoRC**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by MementoRC** [comment]
@conda-forge/help-python, Hello, ready_for_review

**Comment by klpanagi** [comment]
Hello

File: recipes/commlib-py/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+{% set version = "0.10.6" %}
+{% set commit = "b812d91bdad45ceab84003d372683e6fd6985719" %}
+{% set sha256 = "a2ead2622c29481d1cb1382f51bb28ee0963f01df0579d71071f2bd926deface" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/robotics-4-all/commlib-py/archive/{{ commit }}.tar.gz
```
Comment (xhochy) [inline recipes/commlib-py/meta.yaml:11]: Please use the release tag archive.

File: recipes/commlib-py/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    - relaxed-requirements.patch
+    - update_pydantic_test.patch
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install --use-pep517 . -vvv .
+  number: 0
+
+requirements:
+  build:
```
Comment (xhochy) [inline recipes/commlib-py/meta.yaml:23]: This does not need a build section

**Review by xhochy — CHANGES_REQUESTED** [review]

File: recipes/commlib-py/meta.yaml (hunk 1 above)
Comment (MementoRC) [reply recipes/commlib-py/meta.yaml:11]: The owner of the package seems to have discontinued the github releases. I will then revert this recipe to using the pypi package, add the licence and test in the recipe, will this work better?

File: recipes/commlib-py/meta.yaml (hunk 2 above)
Comment (MementoRC) [reply recipes/commlib-py/meta.yaml]: Correct, oversight. Updated

File: recipes/commlib-py/meta.yaml (hunk 1 above)
Comment (MementoRC) [reply recipes/commlib-py/meta.yaml]: Reverted to using the more recent pypi package

File: recipes/commlib-py/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    - rich
+    - ujson >=5.7.0
+
+test:
+  imports:
+    - commlib
+  requires:
+    - coverage
+    - git
+  commands:
```
Comment (xhochy) [inline recipes/commlib-py/meta.yaml:40]: You could use the release tarball from https://github.com/robotics-4-all/commlib-py as the source above and include the tests with
```suggestion
  source_files:
    - test
  commands:
```

Comment (MementoRC) [reply recipes/commlib-py/meta.yaml:48]: The issue was the release version discrepancy between Pypi and the repo release. There was a misunderstanding as to how to use the second source, it is not set with Pypi package (latest version) and tests from github as a second source

**Comment by MementoRC** [comment]
@conda-forge/help-python, Hello, ready_for_review

File: recipes/commlib-py/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+requirements:
+  host:
+    - pip
+    - poetry-core
+    - python >=3.7.0
+  run:
+    - hiredis >=2.1.1
+    - paho-mqtt >=1.6.1
+    - pika >=1.3.1
+    - poetry-core
```
Comment (xhochy) [inline recipes/commlib-py/meta.yaml:26]: Is this really needed? Normally this is a build time only dependency.

Comment (MementoRC) [reply recipes/commlib-py/meta.yaml]: Right, it looks like a absent-minded copy-paste

**Review by xhochy — APPROVED** [review]

**Pull request merged**